examples/client.rs

```diff
@@ -58,14 +58,14 @@ fn main() {
         .map(|_| println!("done"))
         .map_err(|e| println!("error: {:?}", e));
 
-    core.run(done).unwrap();
+    tokio::run(done);
```

Comment:
It looks like you explicitly create a runtime above but then call `tokio::run` here. `tokio::run` internally creates a separate runtime. Doing this will result in 2 sets of threads.

The two options are to call `shutdown_on_idle().wait()` here (which blocks) or pass in a "default executor".

Looking at [`DefaultExecutor`](https://docs.rs/tokio-executor/0.1.2/tokio_executor/struct.DefaultExecutor.html), it looks like it does not implement `Send`, which would mean it does not work here. So, you are stuck w/ option 1.

I wonder if `DefaultExecutor` should impl `Send`. See https://github.com/tokio-rs/tokio/issues/376

Reply:
Yeah, this was the first code I'd written with the new Tokio API and clearly I did not understand it yet at the time. a07b05d3e474fcbcfdb1d3ce32c9d75dd97ed023 fixes this.